src/mui/input/AutocompleteInput.js

```diff
@@ -0,0 +1,119 @@
+import React, { Component, PropTypes } from 'react';
+import AutoComplete from 'material-ui/AutoComplete';
+import title from '../../util/title';
+
+/**
+ * An Input component for an autocomplete field, using an array of objects for the options
+ *
+ * Pass possible options as an array of objects in the 'choices' attribute.
+ *
+ * By default, the options are built from:
+ *  - the 'id' property as the option value,
+ *  - the 'name' property an the option text
+ * @example
+ * const choices = [
+ *    { id: 'M', name: 'Male' },
+ *    { id: 'F', name: 'Female' },
+ * ];
+ * <AutocompleteInput source="gender" choices={choices} />
+ *
+ * You can also customize the properties to use for the option name and value,
+ * thanks to the 'optionText' and 'optionValue' attributes.
+ * @example
+ * const choices = [
+ *    { _id: 123, full_name: 'Leo Tolstoi', sex: 'M' },
+ *    { _id: 456, full_name: 'Jane Austen', sex: 'F' },
+ * ];
+ * <AutocompleteInput source="author_id" choices={choices} optionText="full_name" optionValue="_id" />
+ *
+ * `optionText` also accepts a function, so you can shape the option text at will:
+ * @example
+ * const choices = [
+ *    { id: 123, first_name: 'Leo', last_name: 'Tolstoi' },
+ *    { id: 456, first_name: 'Jane', last_name: 'Austen' },
+ * ];
+ * const optionRenderer = choice => `${choice.first_name} ${choice.last_name}`;
+ * <AutocompleteInput source="author_id" choices={choices} optionText={optionRenderer} />
+ *
+ * You can customize the `filter` function used to filter the results.
+ * By default, it's `AutoComplete.fuzzyFilter`, but you can use any of
+ * the functions provided by `AutoComplete`, or a function of your own
+ * @see http://www.material-ui.com/#/components/auto-complete
+ * @example
+ * import { Edit, AutocompleteInput } from 'admin-on-rest/mui';
+ * import AutoComplete from 'material-ui/AutoComplete';
+ *
+ * export const PostEdit = (props) => (
+ *     <Edit {...props}>
+ *         <AutocompleteInput source="category" filter={AutoComplete.caseInsensitiveFilter} choices={choices} />
+ *     </Edit>
+ * );
+ *
+ * The object passed as `options` props is passed to the material-ui <AutoComplete> component
+ *
+ * @example
+ * <AutocompleteInput source="author_id" options={{ fullWidth: true }} />
+ */
+class AutocompleteInput extends Component {
+    onNewRequest = (chosenRequest, index) => {
```

Comment:
handleNewRequest ? We use onXXX for props and handleXXX for handlers everywhere I think.